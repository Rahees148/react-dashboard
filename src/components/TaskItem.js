import React, { useState } from "react";

//Context
import {useContext} from 'react';
import {TaskContext} from '../context/TaskContext';

import styled from 'styled-components';
import CheckIcon from '../assets/check.svg';
import DeleteIcon from '../assets/trash-solid.svg';
import EditIcon from '../assets/pen-solid.svg';

const Item = styled.div`
  display:flex;
  justify-content:space-around;
  align-items:flex-start;
  padding:20px 0;
  border-bottom:1px solid #E8E8E8;
  &:first-child{
      padding-top:0;
  }
  &:last-child{
    padding-bottom:0;
    border:none;
}
`;
const Action = styled.div`
  margin-left:auto;
  width:55px;
`
const Checkbox = styled.div`
  position:relative;  
  width:20px;
  height:20px;
  border:2px solid #95A4AB;
  border-radius:4px;
  margin-right:12px;
`
const Check = styled.span`
  position:absolute;
  width:20px;
  height:20px;
  top:-5px;
  right:-7px;
  background:url(${CheckIcon}) no-repeat;
  background-size:100% 100%;
`;  

const Text = styled.p`
    color: ${props => props.checked === true ? '#537178' : '#5285EC'};
    text-decoration: ${props => props.checked === true? 'line-through' : 'none'};
    font-size:20px;
    margin-top:-2px;
    line-height:24px;
    font-weight:500;
    text-align:left;
    max-width:calc(100% - 100px);
`;
const Input = styled.input`
  border:1px solid #F4F4F6;
  padding:5px 10px;
  border-radius:4px;
  width:calc(100% - 100px);
`;
const Icon = styled.i`
  margin: 0 5px;
  cursor:pointer;
`

function TaskItem(props){
    
    const [doEdit, setEidt] = useState(false);
    const context = useContext(TaskContext);
    const task = props.task;
    const [checked, setChecked] = useState(task.completed);
    const [taskName, setTaskName] = useState(task.name);

    const handleCheckboxChange = event => {
            setChecked(!checked);
            if(!checked){
                const updatedTask = {...task,
                    completed: true}
                    context.updateItem(updatedTask);
            }else{
                const updatedTask = {...task,
                completed: false}
                context.updateItem(updatedTask);
            }
            
      }

      const updateTask = () =>{
        const updatedTask = {...task,
          name: taskName}
          context.updateItem(updatedTask);
          setEidt(false)
      }
    return (
        
          <Item>
              <Checkbox 
              checked={checked}
              onClick={handleCheckboxChange}
              >
                  {checked  &&
                        <Check />
                    }
                 
              </Checkbox>
              {doEdit ? 
              <Input type="text" value={taskName} autoFocus={true} onChange={e => setTaskName(e.target.value)} onBlur={updateTask} /> :  <Text checked={checked} >{task.name}</Text>}
              <Action>
                  <Icon onClick={() => setEidt(true)} ><img src={EditIcon} alt="Edit" /></Icon>
                  <Icon onClick={() => context.removeItem(task)} ><img src={DeleteIcon} alt="Delete" /></Icon>
              </Action>
          </Item>
          
    )
}

export default TaskItem;