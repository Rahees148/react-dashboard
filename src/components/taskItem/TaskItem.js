import React, { useState } from "react";

import { Style } from "./TaskItem.style";

//Context
import { useContext } from "react";
import { TaskContext } from "../../context/TaskContext";

import DeleteIcon from "../../assets/trash-solid.svg";
import EditIcon from "../../assets/pen-solid.svg";

function TaskItem(props) {
  const [doEdit, setEidt] = useState(false);
  const context = useContext(TaskContext);
  const task = props.task;
  const [checked, setChecked] = useState(task.completed);
  const [taskName, setTaskName] = useState(task.name);

  const handleCheckboxChange = (event) => {
    setChecked(!checked);
    if (!checked) {
      const updatedTask = { ...task, completed: true };
      context.updateItem(updatedTask);
    } else {
      const updatedTask = { ...task, completed: false };
      context.updateItem(updatedTask);
    }
  };

  const updateTask = () => {
    const updatedTask = { ...task, name: taskName };
    context.updateItem(updatedTask);
    setEidt(false);
  };
  return (
    <Style.Item>
      <Style.Checkbox checked={checked} onClick={handleCheckboxChange}>
        {checked && <Style.Check />}
      </Style.Checkbox>
      {doEdit ? (
        <Style.Input
          type="text"
          value={taskName}
          autoFocus={true}
          onChange={(e) => setTaskName(e.target.value)}
          onBlur={updateTask}
        />
      ) : (
        <Style.Text checked={checked}>{task.name}</Style.Text>
      )}
      <Style.Action>
        <Style.Icon onClick={() => setEidt(true)}>
          <img src={EditIcon} alt="Edit" />
        </Style.Icon>
        <Style.Icon onClick={() => context.removeItem(task)}>
          <img src={DeleteIcon} alt="Delete" />
        </Style.Icon>
      </Style.Action>
    </Style.Item>
  );
}

export default TaskItem;
