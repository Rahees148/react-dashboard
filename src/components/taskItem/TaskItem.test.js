
import renderer from 'react-test-renderer';
import TaskItem from './TaskItem';


it('renders correctly', () => {
  const tree = renderer
    .create(<TaskItem key={1} task={{id:1,name:'testing',completed:false,userId: 2}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});