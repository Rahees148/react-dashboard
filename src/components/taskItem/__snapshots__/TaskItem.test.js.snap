// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-bdfBwQ dXjXda"
>
  <div
    checked={false}
    className="sc-dlfnbm dzhhmG"
    onClick={[Function]}
  />
  <p
    checked={false}
    className="sc-eCssSg iAEvQI"
  >
    testing
  </p>
  <div
    className="sc-gsTCUz huGVAz"
  >
    <i
      className="sc-gKsewC ncsNm"
      onClick={[Function]}
    >
      <img
        alt="Edit"
        src="pen-solid.svg"
      />
    </i>
    <i
      className="sc-gKsewC ncsNm"
      onClick={[Function]}
    >
      <img
        alt="Delete"
        src="trash-solid.svg"
      />
    </i>
  </div>
</div>
`;
